modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase

```diff
@@ -0,0 +1,223 @@
+				userLoginEmailAddress = "analyst@testray.com",
+				userLoginFullName = "Analyst User");
+
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the routine page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+		}
+
+		task ("And can see the routine in the table") {
+			AssertElementPresent(
+				key_projectName = ${routineName},
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("And can not see add button") {
+			AssertElementNotPresent(locator1 = "Button#PLUS");
+		}
+
+		task ("When right click in routine") {
+			RightClick.rightClick(
+				key_projectName = ${routineName},
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("Then can not see the 'Edit, Delete' options in right click") {
+			AssertElementNotPresent(
+				item = "Edit",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+
+			AssertElementNotPresent(
+				item = "Delete",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+		}
+
+		task ("And can not see the kebab menu") {
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+
+			AssertElementNotPresent(locator1 = "ObjectAdmin#KEBAB_MENU");
+		}
+	}
+
+	@description = "Story ID: LPS-178269 | Test Summary: Verify that the User with the 'Testray Lead' role should not be able to create/edit/delete the routine but can view"
+	@priority = 3
+	test AssertLeadRoleCanNotCreateEditAndDeleteRoutine {
+		property test.name.skip.portal.instance = "TestrayRoutinePermissions#AssertLeadRoleCanNotCreateEditAndDeleteRoutine";
+
+		task ("Given that the user login with the Testray Lead role") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "lead@testray.com",
+				userLoginFullName = "Lead User");
+
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the routine page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+		}
+
+		task ("And can see the routine in the table") {
+			AssertElementPresent(
+				key_projectName = ${routineName},
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("And can not see add button") {
+			AssertElementNotPresent(locator1 = "Button#PLUS");
+		}
+
+		task ("When right click in routine") {
+			RightClick.rightClick(
+				key_projectName = ${routineName},
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("Then can not see the 'Edit, Delete' options in right click") {
+			AssertElementNotPresent(
+				item = "Edit",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+
+			AssertElementNotPresent(
+				item = "Delete",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+		}
+
+		task ("And can not see the kebab menu") {
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+
+			AssertElementNotPresent(locator1 = "ObjectAdmin#KEBAB_MENU");
+		}
```

Comment:
One test one intention. The last block of the test should be with the `Then` task

Reply:
Done